src/generator/project.ts

```diff
@@ -31,13 +31,45 @@ const gitignores = `.nimbella
 __deployer__.zip
 __pycache__/
 node_modules
+package-lock.json
 .DS_Store
 `
 
+const ignoreForTypescript = 'lib/\n'
+
+// A canned package.json for a minimal typescript project
+const packageJsonForTypescript = `{
+  "main": "lib/hello.js",
+  "devDependencies": {
+    "typescript": "^4"
+  },
+  "scripts": {
+    "build": "tsc -b"
+  }
+}
+`
+
+// A canned tsconfig.json for typescript project
+const tsconfigJSON = `{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "esModuleInterop": true,
+    "importHelpers": true,
+    "module": "commonjs",
+    "outDir": "lib",
+    "rootDir": "src",
```

Comment:
@joshuaauerbachwatson Is there a specific reason for setting rootDir to `src` ?

Reply:
I don't know enough about `tsconfig` semantics to answer this question.   I decided to standardize on the pattern of using a `src` folder for the source and a `lib` folder for the compile results because it is what `oclif` CLI generators use (hence Heroku CLI uses it and so does `nim` CLI in its own builds).   So, I copied a lot of fields from the `oclif`-generated config.   I thought I needed this field if I was putting all the source in  `src`.   If you want to suggest some variations, go for it.

Reply:
@ddebarros I will go ahead and resolve this conversation and merge the PR.  As discussed separately, we are approaching TypeScript support iteratively and I anticipate a refinement pass once we understand how we want to handle larger projects (e.g with multiple actions and shared material).